pkg/apis/core/validation/validation.go

```diff
@@ -3075,6 +3075,39 @@ func validateContainerCommon(ctr *core.Container, volumes map[string]core.Volume
 	allErrs = append(allErrs, validatePullPolicy(ctr.ImagePullPolicy, path.Child("imagePullPolicy"))...)
 	allErrs = append(allErrs, ValidateResourceRequirements(&ctr.Resources, path.Child("resources"), opts)...)
 	allErrs = append(allErrs, ValidateSecurityContext(ctr.SecurityContext, path.Child("securityContext"))...)
+	return allErrs
+}
+
+func validateHostUsers(spec *core.PodSpec, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
```

Comment:
Don't check feature gates in validation... if the field is present, validate it normally.

Instead of checking feature enablement here, drop the field if the feature is disabled in `pkg/api/pod#dropDisabledFields`

See details in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version

Reply:
Thanks! Fixed this now, will be in a few minutes when I push :)